### PR TITLE
Add a section on Cache guide about negative caching and null values.

### PR DESCRIPTION
**Pull request by pilhuhn**
Document some of the findings, working with the cache.

I wanted to also talk about shutting down the noisy stack-trace (https://github.com/quarkusio/quarkus/issues/10804#issuecomment-660911376) but am not able to get this to work.

### CODE REVIEW

File: docs/src/main/asciidoc/cache.adoc [hunk 1]
```diff
@@ -484,5 +490,87 @@ public class CachedService {
     }
 }
 ----
-<1> This method can be used to force a refresh of the cache entry corresponding to the given key. 
+<1> This method can be used to force a refresh of the cache entry corresponding to the given key.
 <2> This method will invalidate all entries from the `foo` and `bar` caches with a single call.
+
+[#negative-cache]
+== Negative chaching and nulls
```
Comment (gastaldi) [inline docs/src/main/asciidoc/cache.adoc:497]:
```suggestion
== Negative caching and nulls
```

File: docs/src/main/asciidoc/cache.adoc [hunk 2]
```diff
@@ -484,5 +490,87 @@ public class CachedService {
...
+
+Sometimes one wants to cache the results of an (expensive) remote call.
+With all remote  calls, they may fail, in which case one may not want to store the bad results.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/cache.adoc:500]: This sentence is confusing, maybe reword a bit?

File: docs/src/main/asciidoc/cache.adoc (hunk 1 above)
Comment (pilhuhn) [reply docs/src/main/asciidoc/cache.adoc]: OOps :)

File: docs/src/main/asciidoc/cache.adoc (hunk 2 above)
Comment (pilhuhn) [reply docs/src/main/asciidoc/cache.adoc]: Yes, you are right. Thanks for the suggestion

**Review by gastaldi — APPROVED** [review]
LGTM, can you please squash your commits? Thanks!

**Comment by pilhuhn** [comment]
Ping - I did the squash

**Comment by gastaldi** [comment]
Thanks!
